snaps_k8s/provision/ansible_configuration.py

```diff
@@ -902,3 +913,159 @@ def delete_weave_interface(k8s_conf):
         }
         ansible_utils.apply_playbook(consts.K8_DELETE_WEAVE_INTERFACE,
                                      variables=pb_vars)
+
+
+def __get_hostname_map(hosts):
+    """Get hostname map function"""
+    logger.info("Argument List:\n hosts: %s", hosts)
+    hostname_map = {}
+    if hosts:
+        for i in range(len(hosts)):
+            hostname = hosts[i].get(consts.HOST_KEY).get('hostname')
+            host_ip = ""
+            ip_addr = hosts[i].get(consts.HOST_KEY).get(consts.IP_KEY)
+            if ip_addr:
+                host_ip = ip_addr
+            hostname_map[hostname] = host_ip
+
+    logger.info('Exit')
+    return hostname_map
+
+
+def __create_host_nodetype_map(hosts):
+    """Get Node types function"""
+    logger.info("Argument List:\n hosts: %s", hosts)
+    hostnode_map = {}
+    for host in hosts:
+        node_type = host.get(consts.HOST_KEY).get(consts.NODE_TYPE_KEY)
+        hostname = host.get(consts.HOST_KEY).get('hostname')
+        hostnode_map[hostname] = node_type
+    logger.info('Exit')
+    return hostnode_map
+
+
+def __ha_configuration(k8s_conf):
+    """HA configuration """
+    loadbalancer_dict = config_utils.get_loadbalancer_dict(k8s_conf)
+    pb_vars = {
+        'SRC_PACKAGE_PATH': config_utils.get_artifact_dir(k8s_conf),
+        'PROJ_ARTIFACT_DIR': config_utils.get_project_artifact_dir(k8s_conf),
+        'Project_name': config_utils.get_project_name(k8s_conf),
+        }
+
+    hosts = k8s_conf.get(consts.K8S_KEY).get(consts.NODE_CONF_KEY)
+    hostname_map = __get_hostname_map(hosts)
+    host_node_type_map = __create_host_nodetype_map(hosts)
+    logger.info('HA CONFIGURING')
+    __launch_ha_loadbalancer_conf(
+        hostname_map=hostname_map, host_node_type_map=host_node_type_map,
+        loadbalancer_dict=loadbalancer_dict, pb_vars=pb_vars)
+
+    __launch_kubespray_ha_configure(
+        hostname_map=hostname_map, host_node_type_map=host_node_type_map,
+        loadbalancer_dict=loadbalancer_dict, pb_vars=pb_vars)
+
+
+def __launch_ha_loadbalancer_conf(hostname_map=None,
+                                  host_node_type_map=None,
+                                  loadbalancer_dict=None,
+                                  pb_vars=None):
+    """
+    function used to call launch_load_balancer
+    :param hostname_map:
+    :param host_node_type_map:
+    :param loadbalancer_dict:
+    :param project_name:
+    :return:
+    """
+    logger.info("Argument List:\n hostname_map: %s\n host_node_type_map: "
+                "%s\n loadbalancer_dict: %s \n pb_vars: %s",
+                hostname_map, host_node_type_map, loadbalancer_dict,
+                pb_vars)
+
+    logger.info("launch_ha_loadbalancer_conf function")
+    host_node_type_map_local = {}
+    lb_port = str(loadbalancer_dict.get("port"))
+    master_ip_list = []
```

Comment:
use config_utils.get_master_node_ips(k8s_config) to obtain this master list; however, this logic doesn't seem correct as it should be the lb_node. When running the HA CI script, it appears to have picked up the correct one though. Please either use the config_utils function to obtain the master IPs or explain this logic to me.

Reply:
implemented in 'ha_ci_scripts' branch